Fix AT TIME ZONE for TIMESTAMP

martint opened the pull request:
It was producing a timestamp(3) with time zone instead of preserving the precision of the provided timestamp

dain commented on presto-main/src/main/java/io/prestosql/operator/scalar/timestamp/AtTimeZone.java:
```diff
@@ -0,0 +1,142 @@
+            @SqlType("varchar(x)") Slice zoneId,
+            @SqlNullable @SqlType("timestamp(targetPrecision) with time zone") Long dummy) // need a dummy value since the type inferencer can't bind type arguments exclusively from return type
+    {
+        // p is guaranteed to be <= 3 because the short representation for TIMESTAMP WITH TIMEZONE has a max precision of 3
```
`p` is not a variable here

dain commented on presto-main/src/main/java/io/prestosql/operator/scalar/timestamp/AtTimeZone.java:
```diff
@@ -0,0 +1,142 @@
+            @SqlType("varchar(x)") Slice zoneId,
+            @SqlNullable @SqlType("timestamp(targetPrecision) with time zone") Long dummy) // need a dummy value since the type inferencer can't bind type arguments exclusively from return type
+    {
+        // p is guaranteed to be > 3 and <= 6, so no need to round
```
same

martint committed:
Fix AT TIME ZONE for TIMESTAMP

AT TIME ZONE converted timestamps to timestamp(3) with time zone. With this
change, the precision is preserved.
